lib/hexpm_web/controllers/web_auth_controller.ex

```diff
@@ -0,0 +1,93 @@
+defmodule HexpmWeb.WebAuthController do
+  use HexpmWeb, :controller
+  @moduledoc false
+
+  alias Hexpm.Accounts.WebAuth
+
+  # Controller for Web Auth, a means of authenticating the cli from the website
+
+  plug :requires_login when action == :show
+
+  def code(conn, params)
+
+  def code(conn, %{"key_name" => key_name}) do
+    {:ok, response} = WebAuth.get_code(key_name)
+    render(conn, :code, response)
+  end
+
+  def code(conn, _params), do: invalid_params(conn)
+
+  def submit(conn, params)
+
+  def submit(conn, %{"user_code" => user_code}) do
+    audit = audit_data(conn)
+    user = conn.assigns.current_user
+
+    case WebAuth.submit(user, user_code, audit) do
+      {:ok, _request} ->
+        conn
+        |> put_status(:ok)
+        |> json(%{"ok" => "ok"})
```

Comment:
It looks like we are returning JSON for a website request?

Reply:
I am going to redirect to a success page on successful verification. The JSON is temporary.